### PR TITLE
Add CI provider introduction details

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:

Adds more details on how a new CI provider is introduced.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/cc @rthallisey @fabiand 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by xpivarc — COMMENTED** [review]
/approve
/hold

File: docs/release.md [hunk 1]
```diff
@@ -39,24 +39,36 @@ The release schedule is build around a few guiding principles:
...
-The assumption is, that the KubeVirt CI provider will be available by KubeVirt's enhcanement freeze.
+The assumption is, that the KubeVirt CI provider will be available by KubeVirt's enhancement freeze.
+
+### Introduction of a new CI Provider
+
+The introduction of a new provider has the following phases:
+1. Provider is created
+2. Provider is published
+3. Provider lands in kubevirt/kubevirt
+4. Periodic and presubmit lanes get introduced for the sigs `compute`, `network`, `storage` and `operator`, where
```
[the remark this answers is not in the document]
Comment (dhiller) [reply docs/release.md:52]: I was trying to describe the fact here that the sig lanes for `compute`, `network`, `storage` and `operator` are added for every new provider, which is currently not the case for the sig-performance lanes.

We currently have only the sig-performance lanes running with one of the providers (which we might call one instance), for both periodics and presubmits. And I am not sure whether we need to run the sig-performance lanes with all three or even four providers. 

@brianmcarey @fabiand WDYT about that?

But you are implicitly pointing out an interesting fact here: we still are missing a strategy of bumping all lanes that have one instance to the latest providers.

**Comment by dhiller** [comment]
@rthallisey @fabiand do you think this is good to go?

**Comment by rthallisey** [comment]
@dhiller yes.
/lgtm

**Pull request merged**